tern/analyze/default/core.py

```diff
@@ -93,6 +95,46 @@ def execute_base(layer_obj, prereqs):
                 listing_key=prereqs.binary), 'error'))
 
 
+def execute_distroless(layer_obj, prereqs):
+    """Given an ImageLayer object, which is reported as 'Distroless
+    find packagesinstalled in the layer:
+        1. Use command_lib's base to to see if there is a method to
+            retrieve the metadata
+        2. If there is, invoke the scripts on the host and process
+            the results
+        3. Add the results to the ImageLayer object"""
+    # Add notices to this layer object
+    origin_layer = 'Layer {}'.format(layer_obj.layer_index)
+    # find the binary listing
+    listing = command_lib.get_base_listing("distroless")
```

Comment:
I don't think you need this: `default_common.get_base_bin(layer)` will return `distroless` which will automatically go through the `execute_base` path.

Reply:
Sorry: It looks like the reason there is a separate distroless execution path is because there is no shell.

Reply:
Yeah, unfortunately distroless has to be a special case.